src/test/java/org/javarosa/core/model/actions/SetValueActionTest.java

```diff
@@ -0,0 +1,96 @@
+package org.javarosa.core.model.actions;
+
+import org.javarosa.core.test.Scenario;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.javarosa.core.test.AnswerDataMatchers.intAnswer;
+import static org.javarosa.core.util.BindBuilderXFormsElement.bind;
+import static org.javarosa.core.util.XFormsElement.body;
+import static org.javarosa.core.util.XFormsElement.head;
+import static org.javarosa.core.util.XFormsElement.html;
+import static org.javarosa.core.util.XFormsElement.input;
+import static org.javarosa.core.util.XFormsElement.mainInstance;
+import static org.javarosa.core.util.XFormsElement.model;
+import static org.javarosa.core.util.XFormsElement.t;
+import static org.javarosa.core.util.XFormsElement.title;
+
+public class SetValueActionTest {
+    /**
+     * Read-only is a display-only concern so it should be possible to use an action to modify the value of a read-only
+     * field.
+     */
+    @Test
+    public void setvalue_setsValueOfReadOnlyField() throws IOException {
+        Scenario scenario = Scenario.init("Setvalue readonly", html(
+            head(
+                title("Setvalue readonly"),
+                model(
+                    mainInstance(t("data id=\"setvalue-readonly\"",
+                        t("readonly-field")
+                    )),
+                    bind("/data/readonly-field").readonly("1"),
+                    t("setvalue event=\"odk-instance-first-load\" ref=\"/data/readonly-field\" value=\"now()\"")
+                )
+            ),
+            body(
+                input("/data/readonly-field")
+            )));
+
+        assertThat(scenario.answerOf("/data/readonly-field"), is(notNullValue()));
```

Comment:
Good to be able to assert that the value is correct here. Maybe inject current time somehow?